automation/test-execution/ansible/filter_plugins/cpu_utils.py

```
@@ -284,6 +284,43 @@ def extract_numa_nodes(lscpu_data):
+    for part in cpu_range_str.split(','):
+        part = part.strip()
+        if not part:
+            continue
+        if '-' in part:
+            try:
+                start, end = part.split('-', 1)
+                all_cpus.extend(range(int(start), int(end) + 1))
+            except ValueError as e:
+                raise AnsibleFilterError(
+                    f"Invalid CPU range format: '{part}' - {e}"
+                )
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)